crates/router/src/core/payments/operations/payment_complete_authorize.rs

```diff
@@ -126,6 +128,19 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Co
             payment_attempt.payment_method_id.clone(),
         )
         .await?;
+        let customer_acceptance: Option<CustomerAcceptance> = request
+            .customer_acceptance
+            .clone()
+            .map(From::from)
+            .or(payment_method_info
+                .clone()
+                .map(|pm| {
+                    pm.customer_acceptance
+                        .parse_value::<CustomerAcceptance>("Customer Acceptance")
+                })
+                .transpose()
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Failed to deserialize to Payment Mandate Reference ")?);
         let token = token.or_else(|| payment_attempt.payment_token.clone());
```

Comment:
Change the error_message to more appropriate literal, also try to trigger this error locally and test